src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.java

```diff
@@ -47,13 +84,31 @@ public void updateOffenderAttendance(AttendanceDto updatedAttendance) {
                         .eventDate(attendanceData.getEventDate())
                         .eventId(attendanceData.getEventId())
                         .bookingId(attendanceData.getOffenderBookingId())
-                        .period(String.valueOf(attendanceData.getPeriod()))
+                        .period(attendanceData.getPeriod())
                         .paid(attendanceData.isPaid())
                         .attended(attendanceData.isAttended())
                         .prisonId(attendanceData.getPrisonId())
                         .absentReason(attendanceData.getAbsentReason())
                         .eventLocationId(attendanceData.getEventLocationId())
+                        .comments(attendanceData.getComments())
                         .build())
                   .collect(Collectors.toSet());
     }
+
+    public AbsentReasonsDto getAbsenceReasons() {
+        final var paidReasons = Set.of(
```

Comment:
I would prefer the enum to know which are the paid and which are the unpaid reasons, i.e. there could be methods on the enum to return the paid and unpaid reasons